src/overlays/effects/ovl_Effect_Ss_Kakera/z_eff_ss_kakera.c

```diff
@@ -8,50 +8,355 @@
+    diff.z = this->pos.z - this->vec.z;
+    distance = sqrtf(SQXYZ(diff));
+    if (distance > 1000.0f) {
+        return 0;
+    }
+    if (this->regs[0] != 0) {
+        if (func_8097E420(this, &diff, distance) == 0) {
+            return 0;
+        }
+        if (func_8097E4B0(this, &diff, distance) == 0) {
+            return 0;
+        }
+        if (func_8097E4F0(this, &diff, distance) == 0) {
+            return 0;
+        }
+        if (func_8097E584(this, &diff, distance) == 0) {
+            return 0;
+        }
+    }
+    if (func_8097E660(this, &diff, distance) == 0) {
+        return 0;
+    } else {
+        return 1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E698.s")
+void func_8097E7E0(EffectSs* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/func_8097E7E0.s")
+    if (this->regs[8] == 0) {
+        if ((((this->regs[4] >> 4) & 1) * 0x10) == 0x10) {
+            if (this->pos.y <= (player->actor.floorHeight - ((this->regs[4] >> 2) & 3))) {
+                this->regs[9] = 0;
+                this->regs[0] = 0;
+                this->regs[4] &= ~0x60;
+                this->accel.z = 0.0f;
+                this->accel.y = 0.0f;
+                this->accel.x = 0.0f;
+                this->velocity.z = 0.0f;
+                this->velocity.y = 0.0f;
+                this->velocity.x = 0.0f;
+                this->regs[5] = this->regs[9];
+                this->regs[1] = this->regs[9];
+            }
+        } else {
+            if (this->pos.y <= ((player->actor.floorHeight - ((this->regs[4] >> 2) & 3)) - 600.0f)) {
+                this->life = 0;
+            }
+        }
+    } else {
+        switch (this->regs[4] & 3) {
+            case 0:
+                this->regs[8] = 0;
+                break;
+            case 1:
+                if ((this->velocity.y < 0.0f) &&
+                    (func_800C5A20(&globalCtx->colCtx, &this->pos, D_8097EB64[(this->regs[4] >> 2) & 3]))) {
+                    this->velocity.x *= func_8097DE30(0.9f, 0.2f);
+                    this->velocity.y *= -0.8f;
+                    this->velocity.z *= func_8097DE30(0.9f, 0.2f);
+                    if (this->regs[8] > 0) {
+                        this->regs[8] -= 1;
+                    }
+                }
+                break;
+            case 2:
+                if (func_800C5A20(&globalCtx->colCtx, &this->pos, D_8097EB64[(this->regs[4] >> 2) & 3])) {
+                    this->regs[8] = 0;
+                }
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_Kakera/EffectSsKakera_Update.s")
+void EffectSsKakera_Update(GlobalContext* globalCtx, u32 index, EffectSs* this) {
+    switch (((this->regs[4] >> 5) & 3) << 5) {
+        case 0x20:
+            this->regs[2] += 1147;
+            this->regs[3] += 313;
```

Comment:
If these were originally in hex, I think those might be preferred because these are rotation angles. Though I could be wrong on that

Reply:
Yes, angles should be in hex. It's easier to tackle them all at once when they're all in the same format. And the devs are pretty inconsistent: these are likely a sensible number in degrees or something.